core/src/main/java/io/questdb/griffin/engine/table/AsyncFilteredRecordCursor.java

```diff
@@ -281,14 +281,15 @@ public void skipRows(Counter rowCount) throws DataUnavailableException {
     @Override
     public void toTop() {
         // Check if we at the top already and there is nothing to do.
-        if (frameIndex == 0 && frameRowIndex == 0) {
+        if (frameIndex == 0 && frameRowIndex == 0 && frameRowCount == 0) {
```

Comment:
The old check was correct while the new one makes this early exit useless. That's because it'll only kick in when the first frame has no rows.

Reply:
I removed this optimization altogether